refactor(linter): improve `eslint/no-void`

shulaoda opened the pull request:
- Add suggestion fix
- Improve the documentation
- Align the `allowAsStatement` option logic

camc314 commented:
### Merge activity

* **May 25, 4:31 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

shulaoda committed:
refactor(linter): improve `eslint/no-void` (#11285)

- Add suggestion fix
- Improve the documentation
- Align the `allowAsStatement` option logic
